src/backend/commands/createas.c

```diff
@@ -611,11 +592,6 @@ intorel_initplan(struct QueryDesc *queryDesc, int eflags)
 	int			attnum;
 	TupleDesc   typeinfo = queryDesc->tupDesc;
 
-	/* If EXPLAIN/QE, skip creating the "into" relation. */
-	if ((eflags & EXEC_FLAG_EXPLAIN_ONLY) ||
```

Comment:
Why we move these checks out?

Reply:
make sanity check first, we can skip stepping into the function for some case.